docs/concepts/projects/search.md

```diff
@@ -18,6 +18,27 @@ However, it may be preferable to have a stronger governance framework for ML ass
 
 ## Lineage
 
-Hopsworks tracks the lineage (or provenance) of ML assets automatically for you. You can see what features are used in which feature view or training dataset. You can see what training dataset was used to train a given model. For assets that are managed outside of Hopsworks, there is support for the explicit definition of lineage dependencies.
+Hopsworks tracks lineage in two separate ways:
+* implicit
+* explicit
+
+Implicit provenance means that the user doesn't have to change his code and the system will determine links between different artifacts based on their usage(read/write) within the context of the same application. 
+
+Explicit provenance has the user, explicitly marking the links between artifacts by performing additional method calls within their code.
+
+Both of the approache have pros and cons. Implicit provenance may add links where there might be none, because maybe the mechanisms are not fine grained enough. For example, you read two feature groups and write two feature groups. 
+Depending on how fine grained the system is and where in the stack the tracking is being done, there might be links added between both read feature groups and both written feature groups, so four links, even though maybe there should only be two links. Perfect implicit provenance would have all components track all interactions and report them in a way that can be easily combined. If this is not possible, you might want to modify some core components, such as the file system and resource manager and collect imperfect implicit provenance from these, such as we do at Hopsworks. 
```

Comment:
Please remember that this documentation is targeting data scientists using the platform, not researchers or even people interested in changing Hopsworks.

This section has also lots of it depends/maybe and so on. IMHO we can be extremely clear with users (data scientists) what are the limitations of implicit provenance on Hopsworks:

- It requires the feature engineering pipelines to run using the compute on Hopsworks.  So if you are running outside Hopsworks it doesn't capture those dependencies.
- If relies on Hopsworks Jobs/Jupyter notebooks to define the boundaries of the links. If users run multiple pipelines in the same application, then the implicit provenance will link feature groups that are not related to each others.

If you are in one of the situations above, then you have to use explicit provenance. Extremely clear.